### PR TITLE
improve tests and add gateway example using CORS

### PR DESCRIPTION
**Pull request by ronenhamias**

### CODE REVIEW

File: services-examples/src/main/java/io/scalecube/services/examples/helloworld/Example1.java [hunk 1]
```diff
@@ -53,8 +53,6 @@ public static void main(String[] args) {
               System.out.println(consumer.message());
             });
 
-    // shut down the nodes
-    seed.shutdown().block();
-    microservices.shutdown().block();
+    Thread.currentThread().join();
```
Comment (artem-v) [inline services-examples/src/main/java/io/scalecube/services/examples/helloworld/Example1.java]: Get a reference to `Microservices` and call `.onShutdown().block()`

File: services-examples/src/main/java/io/scalecube/services/examples/helloworld/Example1.java [hunk 2]
```diff
@@ -22,8 +22,9 @@
    * Start the example.
    *
    * @param args ignored
+   * @throws InterruptedException blocking main.
    */
-  public static void main(String[] args) {
+  public static void main(String[] args) throws InterruptedException {
```
Comment (artem-v) [inline services-examples/src/main/java/io/scalecube/services/examples/helloworld/Example1.java]: Remove `throws InterruptedException`

File: services-examples/src/main/java/io/scalecube/services/examples/helloworld/Example2.java [hunk 3]
```diff
@@ -31,8 +31,9 @@
    * Start the example.
    *
    * @param args ignored
+   * @throws InterruptedException joining main.
    */
-  public static void main(String[] args) {
+  public static void main(String[] args) throws InterruptedException {
```
Comment (artem-v) [inline services-examples/src/main/java/io/scalecube/services/examples/helloworld/Example2.java]: Remove throws InterruptedException

File: services-examples/src/main/java/io/scalecube/services/examples/helloworld/Example2.java [hunk 4]
```diff
@@ -66,9 +66,7 @@ public static void main(String[] args) {
               System.out.println(greeting.message());
             });
 
-    // shut down the nodes
-    seed.shutdown().block();
-    microservices.shutdown().block();
+    Thread.currentThread().join();
```
Comment (artem-v) [inline services-examples/src/main/java/io/scalecube/services/examples/helloworld/Example2.java]: Get a reference to `Microservices` and call `.onShutdown().block()`

File: services-examples/src/main/java/io/scalecube/services/examples/helloworld/Example3.java [hunk 5]
```diff
@@ -52,8 +52,6 @@ public static void main(String[] args) {
         .doOnNext(onNext -> System.out.println(onNext))
         .blockLast();
 
-    // shut down the nodes
-    seed.shutdown().block();
-    microservices.shutdown().block();
+    Thread.currentThread().join();
```
Comment (artem-v) [inline services-examples/src/main/java/io/scalecube/services/examples/helloworld/Example3.java]: Get a reference to `Microservices` and call `.onShutdown().block()`

File: services-examples/src/main/java/io/scalecube/services/examples/helloworld/Example3.java [hunk 6]
```diff
@@ -23,8 +23,9 @@
    * Start the example.
    *
    * @param args ignored
+   * @throws InterruptedException joining main thread.
    */
-  public static void main(String[] args) {
+  public static void main(String[] args) throws InterruptedException {
```
Comment (artem-v) [inline services-examples/src/main/java/io/scalecube/services/examples/helloworld/Example3.java]: Remove throws InterruptedException

File: services-examples/src/main/java/io/scalecube/services/examples/helloworld/GatewayExample.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+public class GatewayExample {
+
+  /**
+   * Example running gateway service (ws,http) and provisioning greeting service.
+   *
+   * @param args none needed.
+   * @throws InterruptedException joining main thread.
+   */
+  public static void main(String[] args) throws InterruptedException {
```
Comment (artem-v) [inline services-examples/src/main/java/io/scalecube/services/examples/helloworld/GatewayExample.java]: Remove throws InterruptedException

File: services/src/main/java/io/scalecube/services/ServiceCall.java [hunk 8]
```diff
@@ -195,7 +195,9 @@ public ServiceCall requestReleaser(Consumer<Object> requestReleaser) {
     return Mono.defer(
         () -> {
           requireNonNull(address, "requestOne address parameter is required and must not be null");
-          requireNonNull(transport, "transport is required and must not be null");
+          requireNonNull(
```
Comment (artem-v) [inline services/src/main/java/io/scalecube/services/ServiceCall.java]: Let's revert this  error msg

File: services-examples/src/main/java/io/scalecube/services/examples/helloworld/GatewayExample.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+    // Execute the services and subscribe to service events
+    service
+        .sayHello("joe")
+        .subscribe(
+            consumer -> {
+              System.out.println(consumer.message());
+            });
+
+    Thread.currentThread().join();
```
Comment (artem-v) [inline services-examples/src/main/java/io/scalecube/services/examples/helloworld/GatewayExample.java]: Get a reference to `Microservices` and call `.onShutdown().block()`

**Review by artem-v — APPROVED** [review]

**Review by artem-v — DISMISSED** [review]
Fix CR  comments

**Review by segabriel — APPROVED** [review]

**Pull request merged**